### PR TITLE
Support PrivilegeChecker in new parser and fix create analyze fail bug

### LINKED ISSUE #4099: Create analyze fail
### Steps to reproduce the behavior (Required)
1. CREATE ANALYZE DATABASE test;

### Expected behavior (Required)
success
### Real behavior (Required)
Unsupported DdlStmt
### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`

### PR DESCRIPTION
**Pull request by HangyuanLiu**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes #4099

### CODE REVIEW

**Review by murphyatwork — APPROVED** [review]

**Comment by Youngwb** [comment]
I don't get where fix the create analyze fail bug

**Comment by HangyuanLiu** [comment]
run starrocks_fe_unittest

**Comment by HangyuanLiu** [comment]
> I don't get where fix the create analyze fail bug

In the 85th line of PrivilegeChecker, except for the ddl in the if, all will report the error Unsupported DdlStmt. This is not correct.

**Comment by HangyuanLiu** [comment]
run starrocks_fe_unittest

**Comment by HangyuanLiu** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 48 / 76 (63.16%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/analysis/DdlStmt.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|3|6|50.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|25|47|53.19%|
|:large_blue_circle:|com/starrocks/sql/analyzer/AnalyzerUtils.java|11|13|84.62%|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/sql/StatementPlanner.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/analysis/CreateViewStmt.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/CTASAnalyzer.java|5|5|100.00%|
|:large_blue_circle:|com/starrocks/analysis/AlterViewStmt.java|1|1|100.00%|

File: fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java [hunk 1]
```diff
@@ -1034,21 +1034,20 @@ private boolean isStatisticsOrAnalyzer(StatementBase statement, ConnectContext c
     }
 
     private boolean supportedByNewPlanner(StatementBase statement, ConnectContext context) {
-        return statement instanceof QueryStmt
-                || statement instanceof InsertStmt
+        return statement instanceof AlterViewStmt
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java:1037]: could we add a method to StatementBase class? then we needn't a lot of ||.

**Review by Youngwb — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java (hunk 1 above)
Comment (HangyuanLiu) [reply fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java:1037]: I prefer the obvious way, which makes it very easy to understand which statements we support and which ones we don't. If this code seems weird here, can I move it into a function and put it under the sql folder, is that ok?

**Review by Seaven — APPROVED** [review]

**Pull request merged**